hotsos/core/plugins/juju/resources.py

```diff
@@ -81,6 +82,14 @@ def version(self):
             except subprocess.CalledProcessError as exc:
                 log.debug("failed to get juju version from %s (%s) - trying "
                           "config", self.agent_bin_path, exc)
+            except OSError as exc:
+                if exc.errno == errno.ENOEXEC:
+                    # Probably a different arch than host where hotsos runs.
+                    # So we get the version from the symlink.
+                    _dir = os.readlink(os.path.dirname(self.agent_bin_path))
+                    return os.path.basename(_dir)
```

Comment:
heh i had not thought of this as a way of getting the version but I guess it does work as well.